src/platform/silabs/wifi/wf200/WifiInterfaceImpl.cpp

```diff
@@ -367,6 +365,73 @@ bool IsStationConnected()
     return wifi_extra.Has(WifiState::kStationConnected);
 }
 
+bool IsHardwareReady()
+{
+    return wifi_extra.Has(WifiState::kStationInit);
+}
+
+CHIP_ERROR TriggerWifiDisconnection(void)
+{
+    ChipLogProgress(DeviceLayer, "STA-Disconnecting");
+
+    sl_status_t status = sl_wfx_send_disconnect_command();
+    VerifyOrReturnError(status == SL_STATUS_OK, CHIP_ERROR_INTERNAL);
+
+    wifi_extra.Clear(WifiState::kStationConnected);
+
+    xEventGroupSetBits(sl_wfx_event_group, SL_WFX_RETRY_CONNECT);
+    return CHIP_NO_ERROR;
+}
+
+CHIP_ERROR GetAccessPointInfo(wfx_wifi_scan_result_t & info)
+{
+    uint32_t signal_strength = 0;
+
+    // TODO: The ap_info.ssid isn't populated anywhere. The returned value is always 0.
+    chip::ByteSpan apSsidSpan(ap_info.ssid, ap_info.ssid_length);
+    chip::MutableByteSpan apSsidMutableSpan(info.ssid, WFX_MAX_SSID_LENGTH);
+    chip::CopySpanToMutableSpan(apSsidSpan, apSsidMutableSpan);
+    info.ssid_length = apSsidMutableSpan.size();
+
+    // TODO: The ap_info.bssid isn't populated anywhere. The returned value is always 0.
+    chip::ByteSpan apBssidSpan(ap_info.bssid, kWifiMacAddressLength);
+    chip::MutableByteSpan apBssidMutableSpan(info.bssid, kWifiMacAddressLength);
+    chip::CopySpanToMutableSpan(apBssidSpan, apBssidMutableSpan);
+
+    info.security = ap_info.security;
+    info.chan     = ap_info.chan;
+
+    sl_status_t status = sl_wfx_get_signal_strength(&signal_strength);
+    VerifyOrReturnError(status == SL_STATUS_OK, CHIP_ERROR_INTERNAL);
+
+    info.rssi = (signal_strength - 220) / 2;
+
+    ChipLogDetail(DeviceLayer, "WIFI:SSID     : %s", ap_info.ssid);
+    ChipLogDetail(DeviceLayer, "WIFI:BSSID    : %02x:%02x:%02x:%02x:%02x:%02x", ap_info.bssid[0], ap_info.bssid[1],
```

Comment:
Im confused with this and the TODO above.. How is ssid/bssid obtained then?

Reply:
I'm not sure how it is supposed to work - I'm assuming when you connect to the AP the idea was to store the information?

The logic around those objects needs to be reworked as a whole - follow up item is filled.